SixTrack/sixtrack.s

```diff
@@ -50382,7 +50437,15 @@ c$$$            endif
             ! Read in particle amplitudes a(part,turn), x,xp,y,yp,sigma,dE/E [mm,mrad,mm,mrad,mm,1]
             ! TODO: This logic breaks apart if there are particle losses;
             !  it is neither handled or checked for...
-            do k=1,fma_nturn(i) !loop over turns
+            ! If normalization within FMA, we now have to always write the full NORM_* file
+            ! Otherwise  one would overwrite the NORM_* file constantly if different FMAs are done
+            ! on the same DUMP file
+            if (dumplast(j) .eq. -1) then
+              dump_last_turn = numl
+            else
+              dump_last_turn = dumplast(j)
+            endif
+            do k=dumpfirst(j),dump_last_turn !loop over turns, use the dump files
```

Comment:
Problem: if dump_last_turn > fma_nturn_max, the array index k will be too big and we'll segfault, even if the DUMP itself is short enough to fit into the array. We need to use an offset either when writing or reading the array.

Also, we're reading the full dump (repeatedly), which means that even if the number of turns we're trying to do the FMA over is small enough, the full dump needs to be < fma_nturn_max.

Reply:
Did you fix this?